dependencies.yaml

```diff
@@ -126,39 +126,45 @@ dependencies:
       - output_types: conda
         packages:
           - &doxygen doxygen=1.9.1
-  cudatoolkit:
+  cuda_version:
     specific:
       - output_types: conda
         matrices:
           - matrix:
               cuda: "11.2"
             packages:
               - cuda-version=11.2
-              - cudatoolkit
           - matrix:
               cuda: "11.4"
             packages:
               - cuda-version=11.4
-              - cudatoolkit
           - matrix:
               cuda: "11.5"
             packages:
               - cuda-version=11.5
-              - cudatoolkit
           - matrix:
               cuda: "11.6"
             packages:
               - cuda-version=11.6
-              - cudatoolkit
           - matrix:
               cuda: "11.8"
             packages:
               - cuda-version=11.8
-              - cudatoolkit
           - matrix:
               cuda: "12.0"
             packages:
               - cuda-version=12.0
+  cuda:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.*"
+            packages:
+              - cudatoolkit
+          - matrix:
+              cuda: "12.*"
+            packages:
```

Comment:
I considered listing `cuda-cudart-dev` here but no packages aside from the compiler are technically needed for CUDA 12. The `cuda-nvcc` package (from the `build` dependency list) will include the `cuda-cudart` pieces needed for runtime. I can't think of a current use of dependencies.yaml where this distinction would matter, but it's possible that we might want to add `cuda-cudart-dev` here in the future for reasons I can't think of at the moment. I decided to leave it out because it would result in a new line for `cuda-cudart-dev` in the `conda/environments/all_cuda-120_arch-x86_64.yaml` file which didn't seem necessary.

Reply:
So in cuDF (or another RAPIDS library), this would include a few more packages

Reply:
Yes, that's correct.